tfhe/src/shortint/list_compression/compression.rs

```diff
@@ -87,8 +87,9 @@ impl CompressionKey {
                     );
 
                     let mut ct = ct.clone();
-
-                    unchecked_scalar_mul_assign(&mut ct, message_modulus.0 as u8);
+                    let max_noise_level =
+                        MaxNoiseLevel::new((ct.noise_level() * message_modulus.0).get());
```

Comment:
this noise comes from "nowhere" right, we have no key for a reference max noise ?

Reply:
@mayeul-zama the key has no information of the multiplication required to compress ?

Reply:
The compression parameters take into account this shift of the message (which is clean, see assert `NoiseLevel::NOMINAL` above)  to the carry bits.
In my understanding, it means that, in this compression pattern contract,  the multiplication required to compress is by `message_modulus`.

@fakub could you confirm?

Reply:
What I mean is, does it have the constant to multiply with in the key ?

Reply:
It's `message_modulus` which is not stored in `CompressionKey`

Reply:
well that's not good at all, means keys can be easily misused

Reply:
When misused, it would most likely fail a few lines above on
```
 assert_eq!(
    lwe_size,
    ct.ct.lwe_size(),
    "All ciphertexts do not have the same lwe size as the packing keyswitch key"
);
```

But indeed it would be safer also check against the `CompressionKey` expected `message_modulus`

Reply:
Not sure I followed everything, is there something to fix in this PR ?

Reply:
not for you but keys are missing an information as far as I can tell